src/crystal/dwarf/line_numbers.cr

```diff
@@ -212,21 +216,54 @@ module Crystal
           sequence.offset = offset
           sequence.unit_length = @io.read_bytes(UInt32)
           sequence.version = @io.read_bytes(UInt16)
+
+          if sequence.version < 2 || sequence.version > 5
+            raise "unknown line table version: #{sequence.version}"
+          end
+
+          if sequence.version >= 5
+            sequence.address_size = @io.read_bytes(UInt8).to_i
+            sequence.segment_selector_size = @io.read_bytes(UInt8).to_i
+          else
+            sequence.address_size = {{ flag?(:bits64) ? 8 : 4 }}
+            sequence.segment_selector_size = 0
+          end
+
           sequence.header_length = @io.read_bytes(UInt32)
-          sequence.minimum_instruction_length = @io.read_byte.not_nil!
+          sequence.minimum_instruction_length = @io.read_bytes(UInt8).to_i
 
           if sequence.version >= 4
-            sequence.maximum_operations_per_instruction = @io.read_byte.not_nil!
+            sequence.maximum_operations_per_instruction = @io.read_bytes(UInt8).to_i
+          else
+            sequence.maximum_operations_per_instruction = 1
           end
 
-          sequence.default_is_stmt = @io.read_byte.not_nil! == 1
-          sequence.line_base = @io.read_bytes(Int8)
-          sequence.line_range = @io.read_byte.not_nil!
-          sequence.opcode_base = @io.read_byte.not_nil!
+          if sequence.maximum_operations_per_instruction == 0
```

Comment:
Do we need those non zero checks here and below, now that DWARF 5 is supported?

Reply:
We could still read compromised data. I think it makes sense to have some basic sanity checks.